packages/compiler/test/expression_parser/parser_spec.ts

```
@@ -210,6 +210,12 @@ describe('parser', () => {
         checkAction('a.a["a"]');
       });
 
+      it('should parse safe keyed reads', () => {
+        checkAction('a?.["a"]');
+        checkAction('this.a?.["a"]', 'a?.["a"]');
```

Could you plz add a case with nested reads like this:
```
this.a?.[this.b?.["a"] ?? 'a']
```

---

I'm not sure what's going on there. Maybe a bug in the expression printer that's used in the test?

---

Was this resolved?

---

I figured out why it was failing: it was asserting using `checkAction` which had a parse error since pipes aren't allowed in actions, but the parse errors are suppressed unless the assertion error goes through `checkActionWithError`.

I've changed it to `checkBinding` and added the pipe case.